apps/dashboard/src/hooks/use-fetch-subscribers.ts

```diff
@@ -0,0 +1,43 @@
+import { getSubscribers } from '@/api/subscribers';
+import { QueryKeys } from '@/utils/query-keys';
+import { keepPreviousData, useQuery } from '@tanstack/react-query';
+import { useEnvironment } from '../context/environment/hooks';
+
+interface UseSubscribersParams {
+  cursor?: string;
+  query?: string;
+  email?: string;
+  phone?: string;
+  name?: string;
+  subscriberId?: string;
+  limit?: number;
+}
+
+export function useFetchSubscribers({
+  cursor = '',
+  query = '',
+  email = '',
+  phone = '',
+  name = '',
+  subscriberId = '',
+  limit = 8,
+}: UseSubscribersParams = {}) {
+  const { currentEnvironment } = useEnvironment();
+
+  const subscribersQuery = useQuery({
+    queryKey: [
+      QueryKeys.fetchSubscribers,
+      currentEnvironment?._id,
```

Comment:
Would suggest to switch places, then we can do cache deletion by pattern of the environment id and it will remove all the child keys in the array aswell

Reply:
We need to discuss about doing this for every query, so please raise it independently.